src/GitLink/Linker.cs

```diff
@@ -41,22 +41,37 @@ public static bool Link(string pdbPath, LinkOptions options = default(LinkOption
             {
                 sourceFiles = pdb.GetFilesAndChecksums().Keys.ToList();
 
-                repositoryDirectory = GitDirFinder.TreeWalkForGitDir(Path.GetDirectoryName(sourceFiles.First()));
-                if (repositoryDirectory == null)
+                if (options.GitWorkingDirectory != null)
```

Comment:
Shouldn't we check for string.IsNullOrWhitespace instead? Or do we want to allow "" ?